packages/collabswarm/src/acl.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * An ACL keeps track of a list of user's public keys and produces changes that
+ * can be sent to other swarm peers.
+ *
+ * @tparam ChangesType Type of a change record.
+ * @tparam PublicKey Type of a user's public key.
+ */
+export interface ACL<ChangesType, PublicKey> {
+  /**
+   * Add a new user to the ACL.
+   *
+   * @param publicKey User's public key.
+   * @return A change record for the addition to the ACL.
+   */
+  add(publicKey: PublicKey): Promise<ChangesType>;
+
+  /**
+   * Remove a user from the ACL.
+   *
+   * @param publicKey User's public key.
+   * @return A change record for the removal from the ACL.
+   */
+  remove(publicKey: PublicKey): Promise<ChangesType>;
+
+  /**
+   * Gets a change record describing the current state of the ACL.
```

Comment:
`change_records sounds like a specific change, not the current list…. Maybe clear from def ChangesType or in the other PR, but if not maybe double check that it seems accurate

Reply:
Ah a change record here is a description of a change to a document where that can be defined as a set of changes merged into a single change record.

In the case of automerge, this type is `BinaryChange[]`
In the case of Y.js, this type is `Uint8Array`

I can make this plural to indicate that these can always contain multiple changes